js.go

```diff
@@ -1127,6 +1162,121 @@ func (js *js) AddConsumer(stream string, cfg *ConsumerConfig) (*ConsumerInfo, er
 	return info.ConsumerInfo, nil
 }
 
+// JSApiConsumerDeleteResponse.
+type JSApiConsumerDeleteResponse struct {
+	APIResponse
+	Success bool `json:"success,omitempty"`
+}
+
+func (js *js) DeleteConsumer(stream, durable string) error {
+	if stream == _EMPTY_ {
+		return ErrStreamNameRequired
+	}
+
+	dcSubj := js.apiSubj(fmt.Sprintf(JSApiConsumerDeleteT, stream, durable))
+	r, err := js.nc.Request(dcSubj, nil, js.wait)
+	if err != nil {
+		return err
+	}
+	var resp JSApiConsumerDeleteResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return err
+	}
+	if resp.Error != nil {
+		return errors.New(resp.Error.Description)
+	}
+	return nil
+}
+
+func (js *js) ConsumerInfo(stream, durable string) (*ConsumerInfo, error) {
+	return js.getConsumerInfo(stream, durable)
+}
+
+// APIPagedRequest includes parameters allowing specific pages to be requests from APIs responding with APIPaged
+type APIPagedRequest struct {
+	Offset int `json:"offset"`
+}
+
+type JSApiConsumersRequest struct {
+	APIPagedRequest
+}
+
+// JSApiConsumerListResponse.
+type JSApiConsumerListResponse struct {
+	APIResponse
+	APIPaged
+	Consumers []*ConsumerInfo `json:"consumers"`
+}
+
+// ConsumerLister fetches pages of ConsumerInfo objects.
+type ConsumerLister struct {
```

Comment:
Looking at the rest of the code, it seem that we don't have any lock here, which is fine, but should we then mention that this is not "thread safe"?